core/types/src/key_request.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+//! SGX key request rust types
+
+use crate::{new_type_wrapper, Attributes, MiscellaneousSelect};
+use mc_sgx_core_sys_types::{
+    sgx_config_svn_t, sgx_cpu_svn_t, sgx_isv_svn_t, sgx_key_id_t, sgx_key_request_t,
+    SGX_CPUSVN_SIZE, SGX_KEYID_SIZE, SGX_KEYPOLICY_CONFIGID, SGX_KEYPOLICY_ISVEXTPRODID,
+    SGX_KEYPOLICY_ISVFAMILYID, SGX_KEYPOLICY_MRENCLAVE, SGX_KEYPOLICY_MRSIGNER,
+    SGX_KEYPOLICY_NOISVPRODID, SGX_KEYSELECT_EINITTOKEN, SGX_KEYSELECT_PROVISION,
+    SGX_KEYSELECT_PROVISION_SEAL, SGX_KEYSELECT_REPORT, SGX_KEYSELECT_SEAL,
+    SGX_KEY_REQUEST_RESERVED2_BYTES,
+};
+use rand::{CryptoRng, RngCore};
+
+#[repr(u16)]
+pub enum KeyName {
+    /// Launch key
+    EnclaveInitializationToken = SGX_KEYSELECT_EINITTOKEN,
+
+    /// Provisioning Key
+    Provision = SGX_KEYSELECT_PROVISION,
+
+    /// Provisioning Seal Key
+    ProvisionSeal = SGX_KEYSELECT_PROVISION_SEAL,
+
+    /// Report Key
+    Report = SGX_KEYSELECT_REPORT,
+
+    /// Seal Key
+    Seal = SGX_KEYSELECT_SEAL,
+}
+
+#[repr(transparent)]
+#[derive(Clone, Hash, PartialEq, Eq, Debug, Copy)]
+pub struct KeyPolicy(u16);
+impl KeyPolicy {
+    /// Derive key using the enclave's ENCLAVE measurement register
+    pub const MRENCLAVE: KeyPolicy = KeyPolicy(SGX_KEYPOLICY_MRENCLAVE);
+
+    /// Derive key using the enclave's SINGER measurement register
+    pub const MRSIGNER: KeyPolicy = KeyPolicy(SGX_KEYPOLICY_MRSIGNER);
+
+    /// Derive key without the enclave's ISVPRODID
+    pub const NO_ISV_PROD_ID: KeyPolicy = KeyPolicy(SGX_KEYPOLICY_NOISVPRODID);
+
+    /// Derive key with the enclave's CONFIGID
+    pub const CONFIG_ID: KeyPolicy = KeyPolicy(SGX_KEYPOLICY_CONFIGID);
+
+    /// Derive key with the enclave's ISVFAMILYID
+    pub const ISV_FAMILY_ID: KeyPolicy = KeyPolicy(SGX_KEYPOLICY_ISVFAMILYID);
+
+    /// Derive key with the enclave's ISVEXTPRODID
+    pub const ISV_EXTENDED_PROD_ID: KeyPolicy = KeyPolicy(SGX_KEYPOLICY_ISVEXTPRODID);
+}
```

Comment:
Look into the `bitflags` crate for this...

Reply:
brought in `bitflags`